src/rules/selector-no-redundant-nesting-selector/index.js

```diff
@@ -15,18 +15,43 @@ export default function(actual) {
     }
 
     root.walkRules(/&/, rule => {
-      const { selector } = rule;
-      // "Ampersand followed by a combinator followed by non-combinator non-ampersand and not the selector end"
-      const regex = /^&\s*[>+~ ]\s*[^>+~ {&]+/;
-
-      if (selector === "&" || regex.test(selector)) {
-        utils.report({
-          ruleName,
-          result,
-          node: rule,
-          message: messages.rejected
+      parseSelector(rule.selector, result, rule, fullSelector => {
+        // "Ampersand followed by a combinator followed by non-combinator non-ampersand and not the selector end"
```

Comment:
@kristerkari i think be good leave him for clarify what we do for other contributors